### PR TITLE
Automation tool to automatically upload caplin's snapshot files to R2

### PR DESCRIPTION
**Pull request by Giulio2002**
Upload beacon snapshots to R2 every week by default

### CODE REVIEW

File: p2p/netutil/net.go [hunk 1]
```diff
@@ -299,10 +305,15 @@ func (s *DistinctNetSet) key(ip net.IP) net.IP {
...
 	var buf bytes.Buffer
 	buf.WriteString("{")
-	keys := make([]string, 0, len(s.members))
-	for k := range s.members {
-		keys = append(keys, k)
+	if s.members == nil {
+		s.members = &sync.Map{}
+		s.buf = make(net.IP, 17)
 	}
+	keys := make([]string, 0, 16)
```
Comment (battlmonstr) [inline p2p/netutil/net.go:311]: where is 16 coming from?

File: p2p/netutil/net.go [hunk 2]
```diff
@@ -299,10 +305,15 @@ func (s *DistinctNetSet) key(ip net.IP) net.IP {
 func (s DistinctNetSet) String() string {
 	var buf bytes.Buffer
 	buf.WriteString("{")
-	keys := make([]string, 0, len(s.members))
-	for k := range s.members {
-		keys = append(keys, k)
+	if s.members == nil {
+		s.members = &sync.Map{}
+		s.buf = make(net.IP, 17)
```
Comment (battlmonstr) [inline p2p/netutil/net.go]:
```suggestion
        return "{}"
```
if the map doesn't exist - return an empty representation

File: p2p/netutil/net.go [hunk 3]
```diff
@@ -221,17 +222,21 @@ type DistinctNetSet struct {
...
 // Add adds an IP address to the set. It returns false (and doesn't add the IP) if the
 // number of existing IPs in the defined range exceeds the limit.
 func (s *DistinctNetSet) Add(ip net.IP) bool {
 	key := s.key(ip)
-	n := s.members[string(key)]
+	nI, ok := s.members.Load(string(key))
+	n := uint(0)
+	if ok {
+		n = nI.(uint)
+	}
```
Comment (battlmonstr) [inline p2p/netutil/net.go:237]:
```suggestion
	n := uint(0)
	if value, ok := s.members.Load(string(key)); ok {
		n = value.(uint)
	}
```

**Review by battlmonstr — CHANGES_REQUESTED** [review]

File: p2p/netutil/net.go (hunk 1 above)
Comment (Giulio2002) [reply p2p/netutil/net.go:311]: we can just define it as empty []string

**Review by battlmonstr — APPROVED** [review]

**Pull request merged**